src/utils/section-handler.js

```diff
@@ -0,0 +1,65 @@
+/*
+ * Copyright 2019 Adobe. All rights reserved.
+ * This file is licensed to you under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License. You may obtain a copy
+ * of the License at http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under
+ * the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR REPRESENTATIONS
+ * OF ANY KIND, either express or implied. See the License for the specific language
+ * governing permissions and limitations under the License.
+ */
+
+const fallback = require('mdast-util-to-hast/lib/handlers/root');
+const all = require('mdast-util-to-hast/lib/all');
+const wrap = require('mdast-util-to-hast/lib/wrap');
+
+/**
+ * Get the tag name for the specified section.
+ *
+ * @param {Node} section The MDAST section to get the tag name for
+ * @returns {string} The tag name for the section. Defaults to {@code div}.
+ */
+function getTageName(section) {
+  return (section.meta && section.meta.tagName) || 'div';
+}
+
+/**
+ * Get the class name for the specified section.
+ *
+ * @param {Node} section The MDAST section to get the class name for
+ * @returns {string} The class name for the section. Defaults to {@code hlx-section}.
+ */
+function getClass(section) {
+  return (section.meta && section.meta.className) || 'hlx-section';
+}
+
+/**
+ * Get the types for the specified section.
+ *
+ * @param {Node} section The MDAST section to get the types for
+ * @returns {string} A space-separated list of section types, or {@code null} if none desired.
+ */
+function getTypes(node) {
+  const outputTags = node.meta && node.meta.types === true;
+  const types = (outputTags && node.types) || null;
+  return types ? types.join(' ') : null;
+}
+
+function sectionHandler() {
+  return function handler(h, node, parent) {
+    const n = Object.assign({}, node);
+
+    // we have a section that is not the document root, so wrap it in the desired tag
+    if (parent && parent.type === 'root' && parent.children.length > 1) {
+      const tagName = getTageName(n);
+      const props = { class: getClass(n), 'data-hlx-types': getTypes(n) };
```

Comment:
I'd rather merge the `types` into `class` and then have `data-<prop` for each additional `prop` that's in the section metadata.

Reply:
@trieloff So, something like this?
```
<div class="hlx-Section hlx-Section--hasImage hlx-Section--hasText" data-hlx-<meta1>="<value1>" data-hlx-<meta2>="<value2>" …></div>
```

And we let the dev:
- overwrite the section class
- toggle whether type classes are added (and use the custom section class as prefix if set)
- toggle whether additional meta props are exposed as data attributes

Reply:
@trieloff So, something like this?
```
<div class="hlx-Section hlx-Section--hasImage hlx-Section--hasText" data-hlx-<meta1>="<value1>" data-hlx-<meta2>="<value2>" …></div>
```

And we let the dev:
- overwrite the section class
- toggle whether type classes are added (and use the custom section class as prefix if set)
- toggle whether additional meta props are exposed as data attributes

Reply:
Yes, except that classnames should be lowercase and should be kept as they are (your lower example).

We can fix the collisions later, i.e. through a step that performs class renaming.

My reason for this is that the HTML is simpler, and writing CSS selectors for attributes is harder than for classes, so I'd err on the side of simplicity over consistency.

Reply:
@trieloff If we assume a step that does the renaming, then does it make sense to let the author specify the section class name in the first place in the yaml? I could just hardcode `hlx-section` then.

Less logic => fewer risk of bugs

Reply:
I think allowing to specify the class name is a useful quick fix for some scenarios.

Reply:
i don't think we need to prefix the `section` with `hlx-section` as i think that the entire default DOM is a helix DOM and in case of a conflict the or some other desire to change the DOM it is very easily done in a the `pre.js` via jquery or regular DOM manipulation.